kernel/src/sched/priority_scheduler.rs

```diff
@@ -256,7 +267,7 @@ trait PreemptSchedInfo {
 
     fn cpu(&self) -> &AtomicCpuId;
 
-    fn cpu_affinity(&self) -> &CpuSet;
+    fn cpu_affinity(&self) -> CpuSet;
```

Comment:
I do noticed the cost. The change is because I didn't find a way to borrow a value from the temporary `Arc<Thread>` got from `task.downcast_ref.upgrade.unwrap`. The proposal below naturally solve this problem.